Enhanced Monitoring Doc with some small details (#12117)
kyma-project/kyma: docs/01-overview/main-areas/observability/obsv-03-istio-monitoring.md

The Istio-related instance is a Deployment named `monitoring-prometheus-istio-server`, configured with a short data retention time and hardcoded configuration that you should not change. It also has no PersistentVolume attached. This instance never discovers additional metric endpoints from such resources as ServiceMonitors.

The monitoring chart is configured in such a way that it is possible to scrape metrics using [`Strict mTLS`]
(https://istio.io/latest/docs/tasks/security/authentication/authn-policy/#globally-enabling-istio-mutual-tls-in-strict-mode). For this to work, Prometheus is configured to scrape metrics using Istio certificates. Prometheus is deployed with a sidecar proxy which rotates SDS certificates and outputs them to a volume mounted to the 

> corresponding Prometheus container. To stick to Istio's observability best practices, Prometheus's Istio-proxy is configured to not intercept or redirect any traffic. By default, metrics from Kyma components are scraped using mTLS besides selected components (i.e. Kiali, Prometheus Operator, Prometheus Istio) due to several reasons.

Reviewer: ```suggestion
corresponding Prometheus container. To stick to Istio's observability best practices, Prometheus's Istio-proxy is configured to not intercept or redirect any traffic. By default, metrics from Kyma components are scraped using mTLS with some exceptions (i.e. Kiali, Prometheus Operator, Prometheus Istio) due to several reasons.
```
